SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_GimmighoulChestFarm.cpp

```diff
@@ -0,0 +1,172 @@
+		return std::unique_ptr<StatsTracker>(new Stats());
+	}
+
+    GimmighoulChestFarm::GimmighoulChestFarm()
+        : PP(
+            "<b>First Attack PP:</b><br>The amount of PP remaining on your lead's first attack.",
+            LockWhileRunning::LOCKED,
+            15
+        )
+        , GO_HOME_WHEN_DONE(false)
+        , FIX_TIME_WHEN_DONE(
+            "<b>Fix time when done:</b><br>Fix the time after the program finishes.",
+            LockWhileRunning::LOCKED, false
+        )
+		, NOTIFICATION_STATUS_UPDATE("Status Update", true, false, std::chrono::seconds(3600))
+        , NOTIFICATIONS({
+			&NOTIFICATION_STATUS_UPDATE,
+            &NOTIFICATION_PROGRAM_FINISH,
+            &NOTIFICATION_ERROR_FATAL,
+            })
+    {
+        PA_ADD_OPTION(PP);
+        PA_ADD_OPTION(GO_HOME_WHEN_DONE);
+        PA_ADD_OPTION(FIX_TIME_WHEN_DONE);
+        PA_ADD_OPTION(NOTIFICATIONS);
+    }
+
+    void GimmighoulChestFarm::program(SingleSwitchProgramEnvironment& env, BotBaseContext& context) {
+
+		GimmighoulChestFarm_Descriptor::Stats& stats = env.current_stats<GimmighoulChestFarm_Descriptor::Stats>();
+		uint32_t c = 0;
+		while(c < PP) {
+            //Press A to enter battle, assuming there is a chest
+            env.log("Fetch Attempts: " + tostr_u_commas(c));
+            pbf_mash_button(context, BUTTON_A, 90);
+
+			//Wait for the battle to load then check for battle menu, if there isn't a battle menu then no chest
+			BattleMenuWatcher battle_menu(COLOR_YELLOW);
+			int ret = wait_until(
+				env.console, context,
+				std::chrono::seconds(10),
+				{ battle_menu }
+			);
+
+			if (ret == 0) {
+				//Attack using your first move
+				pbf_mash_button(context, BUTTON_A, 90);
+				c++;
+				context.wait_for_all_requests();
+				OverworldWatcher overworld(COLOR_RED);
+				int ret2 = wait_until(
+					env.console, context,
+					std::chrono::seconds(120),
+					{ overworld }
+				);
+				if (ret2 != 0) {
+					stats.errors++;
+					env.update_stats();
+					throw OperationFailedException(env.console, "Failed to return to Overworld after two minutes. Did your attack miss or fail to defeat Gimmighoul in one hit?");
+				}
+				stats.pokemon_fainted++;
+				env.update_stats();
+				send_program_status_notification(env, NOTIFICATION_STATUS_UPDATE);
+
+				//Walk forward since battles force you to jump back
+				pbf_press_button(context, BUTTON_L, 50, 40);
+				pbf_move_left_joystick(context, 128, 0, 120, 0);
+			}
+
+            //Close the game
+			save_game_from_overworld(env.console, context);
+            pbf_press_button(context, BUTTON_HOME, 20, GameSettings::instance().GAME_TO_HOME_DELAY);
+            context.wait_for_all_requests();
+            close_game(context);
+			
+            //Date skip - in-game day cycle is 72 mins, so 2 hours is fastest way
+			//This isn't perfect because 12 hour format but it works
+            home_to_date_time(context, true, false);
```

Comment:
hmm, I think this will depends on your clock format. For which format is it ? Maybe Kuro or Gin know a function to handle every clock format ?

Reply:
experimented with changing the region/clock format on my switch, it works for both 12 and 24 hour clock layouts (if anything the 24 hour format runs more efficiently). if there's another layout other than those two I'll have to make changes.

12 hour format appears in the settings as month/day/year hour:minute am/pm okbutton.
24 hour format appears in the settings as day/month/year hour:minute okbutton.

hour is in the same position for both formats so moving it forward is fine in both formats. to save the change it presses A 4 times to navigate past minute and am/pm and to press the confirm button. 24 hour format doesn't have am/pm so it ends up pressing A an extra time after it saves the change, which just reopens the date/time menu. the home button press afterwards closes everything and takes it back to menu in both formats.